### ISSUE TITLE
Recipes take more charges than batteries can hold

### ISSUE THREAD

**Issue by MochaModifier** [CleverRaven/Cataclysm-DDA#64275]
### Describe the bug

Various recipes, particularly food handling ones, require more charges (or  over half the capacity) of a standard rechargeable medium battery.

While it may be impractical to try and run an oven off just a car battery - portable cooking devices should be able to _cook_ using the batteries they are designed to use. You should be able to make more than a single fruit leather before needing to reload your hotplate, messkit, or even the supposedly "more efficient" induction hot plate.

### Attach save file

N/A

### Steps to reproduce

1. Try to cook or craft anything with a battery operated portable cooker, or cooking capable device
2. With batteries having 500 (or the rarer 600 charge 'high capacity' version) - you'll be lucky if you can make a single pancake.

### Expected behavior

Either batteries should hold more charge, or devices should be capable of performing tasks they're designed for without requiring a pile of batteries and a charging bank; just so you can make breakfast for the day.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.2604 (21H2)
- Game Version: 0894971 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Bionic Professions [package_bionic_professions],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    SpeedyDex [speedydex],
    Tamable Wildlife [Tamable_Wildlife],
    Extra Mutated Scenarios [extra_mut_scens],
    Translate Complex Dialogue [translate_dialogue],
    Alternative Map Key [alt_map_key],
    No Rail Stations [No_Rail_Stations]
]

### Additional context

Never mind portable options, with multiple solar panels and turbines going you can deplete multiple large batteries just making a few meals. 

This isn't something intensive like chemistry or the like - it's simply dehydrating fruit, or baking bread.

An electric hotplate runs around 1k to 3k watts, while a Microwave Oven uses roughly 17k watts. At best, 6x as much power to boil the same cup of water as a hotplate; yet supposedly it's more power efficient to make my pancakes in a microwave?!

I will point out that electric stoves and hotplates both are designed to cook or bake without running their power constantly - either heating up coils or compartments which are designed to retain heat, and heat up cooking utensils like pans or pots in turn. You can easily see this by turning such a device on in real live, and watching their coil go red hot and cool off in cycles.

Please re-evaluate. I've had to leave dehydrated fruit 'in progress' because even after reloading the **purpose built food dehydrator** 6 times; I still don't have the 4 dehydrated apricots I was trying to make! 
![image](https://user-images.githubusercontent.com/38334602/225260220-c940fd4a-8128-4f4e-8628-4c7e824a943e.png)

Even if I wasn't using rechargeable batteries, the disposable variants have no where near the capacity to do anything meaningful.

**Comment by Karol1223** [comment]
The problem is that you can't start the craft if the battery doesn't have enough charge, not that the charges required are too much. Portable cooking appliances like hotplates are designed to be plugged in, not run on batteries, hence our current stupid versions of them don't really get the job done (as they shouldn't). The player should be able to start the craft with a warning prompt then get interrupted when the battery charge runs out so they can reload (which is already the case in some recipes, I've encountered this behaviour a lot when making flour pre-cooking cost audit)

**Comment by MochaModifier** [comment]
And do we even have those appliace options to plug into a gride, or is this a situation where the cart's gone before the horse?

I'm looking in the construction menu now, and I don't see anything about chemistry sets, hotplates, and from another issue - we apparently don't even have microwaves working yet.
Odder still, I apparently can't use the atomic coffee maker now - because recipes **require** charges, but the device **can't** have charges.

All this has done, is make using these devices incredibly inconvenient, or flatly broken them.

**Comment by Karol1223** [comment]
> I thought that maybe just placing the decides near revealed wires might do it - but I still can't cook so much as toast.

Placing them near revealed wires was my suggestion for how it works and I've been saying this for months. Unfortunately that needs someone to code it in.

>All this accomplishes without appliance options, is make using these devices incredibly inconvenient, (or flatly broken them.)
This seems like a change that could have waited.

It waited for months. Anyone could have reviewed it and pointed these things out. This is early experimental we're in, these sort of situations are not only common but also not particularly frowned upon - when something doesn't work the odds of someone fixing it (theoretically) increase. I say theoretically because in many cases this just discourages people from playing. Personally to me that makes it a better incentive to use appliances/vehicle parts as opposed to item forms.

**Comment by flybye3o6** [comment]
> And do we even have those appliance options to plug into a grid, or is this a situation where the cart's gone before the horse?

that is the nature of experimental
plans are to have crafts at lest start even without energy requirements met so you can swap out batteries or generate more power along the way
also to have extension cables plug into electric items and general ways to have items plug into a grid, i dont think anyone is working on this yet

**Comment by MochaModifier** [comment]
I totally get that this is experimental - that's why I play it, and why I post about issues here as I find them.
And I guess I am now pointing this issue out, because the battery cost is so ridiculous - I'm not even sure why we've left it an option.

Never mind realism - if these things couldn't cook on a single charge pre-cataclysm, no one in that world would have bought them, let alone bought enough of them to be common place.
I totally agree - these being plugged in would make total sense.

My issue now, after your posts though, has me worried this change was made more to annoy people into making a desired change.
Not to improve the game directly, or affect a positive change to the gameplay loop.

**Comment by Karol1223** [comment]
It wasn't made to improve the game, it was made to make the cooking costs realistic. These don't go hand in hand. People famously cry when things become harder.

**Comment by flybye3o6** [comment]
i actually like the idea that the energy costs are more accurate, this is a survival game managing resources is what its all about, up until now power was never an issue for me just put down a car battery and a few solar panels, never concerned myself with rationing power when it came to crafting

**Comment by MochaModifier** [comment]
I wouldn't describe this as being harder, and more needlessly inconveniencing. What challenge do you proprose is there, in having to reload a hotplate every few % of progress on a recipe?

I could see if there was a way to automatically reload devices with batteries, so long as they were to hand. 

Again, I'm not complaining because this seemed like a balancing issue with regards to how the devices function.
And again
>when something doesn't work the odds of someone fixing it (theoretically) increase
This is why I said what I did with regard to annoying people to get them to make a change to the game; considering what this 'realism' change has accomplished

**Comment by Zireael07** [comment]
@MochaModifier It's not about challenge, it's about realism. Electric cookers, whether radiant or inductions, have a HUGE power draw indeed. I doubt they could ever work on batteries - what is in the game is legacy from before the rework and should likely be removed
Also what you added in an edit already is a thing in Cata.

**Comment by Karol1223** [comment]
Anid has not only recalculated the cost of every single cooking recipe but also added notes accordingly under most of them. They've poured tons of work into this PR. Don't undermine this by calling it 'made to inconvenience'. They've done tons of work and it is not possible for a single person to catch every problem with a PR - much less so when the PR has like 5k lines changed like this one did. Again, there've been months for review for this, nobody but one or two people bothered.

It's merged now. Someone can go and make sure batteries get audited as well. Or maybe someone will instead make items able to be plugged into the grid. Or maybe neither, it all depends on whether we get someone interested in the matter. In the end, there isn't a problem with battery charge or batteries - it's a problem with these items using batteries.

**Comment by MochaModifier** [comment]
> @MochaModifier It's not about challenge, it's about realism. Electric cookers, whether radiant or inductions, have a HUGE power draw indeed. I doubt they could ever work on batteries - what is in the game is legacy from before the rework and should likely be removed Also what you added in an edit already is a thing in Cata.

And I suppose my issue is again - WHERE is the change that adds the necessary devices then? This change appears to make existing battery operated devices unusably inconvenient - consuming tens of batteries to do anything meaningful.

No care for explaining why this was needed beyond 'realism' - in the game where near future high tech exists.
No ones attempted to explain or point out how this realism has accounted for charge used versus the power cost to cook meals, or questions like efficiency.

I tried to open that discussion - yet people seeem to be sticking on the higher costs being more realistic because batteries.

This also comes alongside other being added, such as the induction hotplate- whose description claims to make using batteries possible by being more efficient than a standard hotplate.

 I am not the one concerned about challenge, I am the one being accused of 'complaining the game is harder'. I'd suggest discussing the issue directly instead.

**Comment by MochaModifier** [comment]
> Anid has not only recalculated the cost of every single cooking recipe but also added notes accordingly under most of them. They've poured tons of work into this PR. Don't undermine this by calling it 'made to inconvenience'. They've done tons of work and it is not possible for a single person to catch every problem with a PR - much less so when the PR has like 5k lines changed like this one did. Again, there've been months for review for this, nobody but one or two people bothered.
> 
> It's merged now. Someone can go and make sure batteries get audited as well. Or maybe someone will instead make items able to be plugged into the grid. Or maybe neither, it all depends on whether we get someone interested in the matter. In the end, there isn't a problem with battery charge or batteries - it's a problem with these items using batteries.
> 
> I was already suggested to remove the hotplate from being able to be used until items can be plugged into the grid similar to how TVs (don't) work right now. I opted not to, for now, but this issue makes me believe that was not the right call here. The mess kit is losing the ability to use batteries the second my next PR gets merged. Several others seem to need to be on my radar now as well.

I don't mean offense to Anid or their work, I merely responded to how you'd described the situation; because that's where the impression came from.
For future reference, It would have been more productive to instead say this up front, and link me to their work - it would be easier to give constructive feedback then

**Comment by Zireael07** [comment]
> And I suppose my issue is again - WHERE is the change that adds the necessary devices then?  

Not made yet - this is what Karol was telling you, that this is the nature of experimental
 
>  in the game where near future high tech exists

Current game is set in like tomorrow with pretty much no high tech except labs. Again, you're thinking of the past versions which were set around 2050 iirc and where a lot of stuff could be explained away with 'but future'

> induction hotplate- whose description claims to make using batteries possible by being more efficient than a standard hotplate.
  
Description needs to be reworked then - yes it is more efficient, but still NOWHERE close to allowing it to run on batteries

Also re your battery concerns, there's #63978 which will rework the charges/power again to make it less of a mess

**Comment by MochaModifier** [comment]
> > And I suppose my issue is again - WHERE is the change that adds the necessary devices then?
> 
> Not made yet - this is what Karol was telling you, that this is the nature of experimental
> 
> > in the game where near future high tech exists
> 
> Current game is set in like tomorrow with pretty much no high tech except labs. Again, you're thinking of the past versions which were set around 2050 iirc and where a lot of stuff could be explained away with 'but future'
> 
> > induction hotplate- whose description claims to make using batteries possible by being more efficient than a standard hotplate.
> 
> Description needs to be reworked then - yes it is more efficient, but still NOWHERE close to allowing it to run on batteries
> 
> Also re your battery concerns, there's #63978 which will rework the charges/power again to make it less of a mess

I suppose then the other thing - you very much can use a multicooker off batteries; but it's a question of what batteries.
I say this more to move this back to a productive discussion - because here's how it can work.

You can look up now, something called a 'solar generator'. These are portable, solar powered "generators" with an inverter to power AC devices. The bulk of these devices is simply a battery - typically ranging from 300Wh to 1000Wh; powerful enough to run a hotplate continuously for days.
These are roughly the size of a toolbox or car battery - and need to be plugged into a solar panel. They come with US style outlets, USB ports etc.

You can run anything on batteries if they're big enough. And they don't have to be _that_ big to work either.

**Comment by Zireael07** [comment]
Google says a "hotplate" (to use the exact term from the game) draws 1000-1200 W, which fits with my experience of induction cooker (my two pot cooker draws around 2500W). So the 1000Wh battery will only last you for an hour. Not quite "days" you're envisioning

Yeah I know lots of things can be run off a car battery, I sometimes run my computer off one (or the fridge) if there's a power outage. Also, even that average car battery, non-electric, is HEAVY as all #$%@#

**Comment by MochaModifier** [comment]
Yeah, I mixed the math a bit there - you could do that with the electric car battery. 
But so we're clear about things - electric cars and most rechargeable batteries use lithium cells (or NiMH). 
A 'traditional' car batter is a lead acid battery.

When I'm talking about the solar generator - we're talking about something about the size of a car battery, but with a greater energy density - which I presume is what storage batteries are meant to represent given where you tend to find them.

So, my wondering is if current storage batteries have too little a capacity - because you could run a hotplate on one for a long time.

**Comment by Zireael07** [comment]
Yeah, a solar generator would definitely work.

No clue on what storage batteries are except they likely are li-ion.

Modern electric car batteries apparently vary WIDLY. https://content.green.car/guides/electric-car-kw-figures-explained/ says from 40 kWh to 94 kWh, but Wiki says an electric hummer has a battery of a whooping 213 kWh 

Even a 40 kWh battery from a Nissan Leaf would indeed work for a looong time, BUT ... https://electronics.stackexchange.com/questions/496625/what-is-the-battery-weight-of-the-nissan-leaf-62-kwh note that the 40 kWh version weights 303 kg, so you aren't moving THAT in any way short of somehow rigging connections to the actual car.

TLDR the only way you can run a hotplate on a battery is on a modern electric car battery, which requires connecting to the car itself because it's too HEEEAVY to be moved independently, so the only way you could do it in CDDA terms is to use the vehicle system, not the crafting system itself

**Comment by Termineitor244** [comment]
Interesting discussion in the comments, but I think some things need to be told again with regards to this issue:

/confirmed because this is indeed a problem. Devices made to cook food should either:
A) Use batteries big enough for their crafts.
B) Be usable only when connected to a grid.
C) Having the ability to start the crafts even without all the necessary charges for the craft in question.

This issue could be resolved by:

A) Expanding the battery capacity for items using medium batteries (For devices in which batteries can actually be used in real life).
B) Adding appliances versions of these items so they can be plugged into the grid (Or using the suggestion of Carol of making the items work when dropped in the vicinity of the grid).
C) Making the necessary changes so the crafts can be started even without full charge, but this is only a band aid in my opinion. 

Feel free to correct me or add more possible solutions to this issue if they solve the underlying problem, but I think that's it.

**Comment by Zireael07** [comment]
A) doesn't apply to hotplates I think. There are NO non-electric car batteries that would work, even a typical lead-acid car battery is a no-go since even a high-end 1000Wh only last you an hour - unless we assume medium battery is 1000Wh (but then what is a large battery?) AND no food crafts are longer than an hour, but I believe there are crafts that take more than 1h

I think B) *is* the direction to go currently and C) would be a nice thing to have for laters

**Comment by Karol1223** [comment]
99% of the affected items should not be using batteries in the first place so there's really one obvious fix to be made here and I'll stand by that.

**Comment by MochaModifier** [comment]
Best keep coffee makers on the list of possible battery operated devices atleast.
These do exist atleast; for real examples: Makita, Coleman and other manufacturers make these; brewing a couple cups of coffee on a single charge.
Some are hybrid devices, which can run on batteries or be plugged into an outlet.

**Comment by flybye3o6** [comment]
coffee makers have a smart design that uses pressure to assist with boiling they are more energy efficient then kettles
idk if there is a way to have that work that way in recipes

**Comment by anoobindisguise** [comment]
You're too focused on still trying to use a battery hotplate. You can easily find tanks of propane with 10s of thousands of ml of propane, one of those will last you ages. Or use charcoal. Or start a furniture fire because fires are still "free" energy. A coffeepot being able to make maybe 3 cups of coffee on a battery is true to life, they were meant to be ran on a power grid. You can make those tools use heavy batteries if you want though

**Comment by MochaModifier** [comment]
> You're too focused on still trying to use a battery hotplate. You can easily find tanks of propane with 10s of thousands of ml of propane, one of those will last you ages. Or use charcoal. Or start a furniture fire because fires are still "free" energy. A coffeepot being able to make maybe 3 cups of coffee on a battery is true to life, they were meant to be ran on a power grid. You can make those tools use heavy batteries if you want though

I think we've already discussed this to the point that we're past it.

**Comment by Zireael07** [comment]
@KHeket that looks like the boiled bandage recipe is missing batch craft savings (realistically you'd just dump them all at the same time so you'd need the same amount of power)

Speaking of your comparison to a car "accumulator", a typical car battery (according to Google) has around 1200-1300Wh so... "one hour of hotplate usage" that I stated above still applies. The recipe seems to be 20 minutes long so the numbers seem to be in roughly the right ballpark. (Also: #63978 is going to rework batteries anyway)

**Comment by Terrorforge** [comment]
For what it's worth, most of these devices DO have grid-powered versions. An oven or onboard chemistry kit does all the work of a hotplate, the foodco kitchen buddy can be placed as an appliance and provides a dehydrator, etc. So you _can_ run most anything off electricity, provided you have enough batteries and solar panels.

But a lot of those things are really roundabout and/or non-obvious. You shouldn't have to build the kitchen buddy to use a dehydrator, you should just be able to plug it in.

**Comment by Zireael07** [comment]
> You shouldn't have to build the kitchen buddy to use a dehydrator, you should just be able to plug it in.

Yep, I expect a PR soon-ish will make appliance versions of those

**Comment by KHeket** [comment]
> Feel free to correct me or add more possible solutions to this issue if they solve the underlying problem, but I think that's it.

I found out one more solution. Leave unrealistic battery system and battery devices for aftershock due the "high tech future super batteries" 
If you want to play hard realistic - play vanila. If you want more sci-fi - play aftershock
So dont fully remove existing batterys and functions of items - just transfer it to aftershock

**Comment by Karol1223** [comment]
Aftershock is not a dumping ground for unrealistic garbage, that's something you should be consulting with Aftershock Maintainers.

**Comment by irwiss** [comment]
Fixed by https://github.com/CleverRaven/Cataclysm-DDA/pull/64682 + https://github.com/CleverRaven/Cataclysm-DDA/pull/65112

**Comment by RenechCDDA** [comment]
It's also worthwhile to note that there are no longer hotplates which take batteries, something which was at odds with reality.

**Comment by fire1919ua** [comment]
An absurd example in the latest experimental build:
-To boil 1L of water you need 17minutes and multi-coocker with 800 points of energy
-To create pickaxe you need 7hours and electric forge with 40 points of energy
So you need less energy to melt steel than to boil water.

**Comment by Zireael07** [comment]
O_o

**Comment by irwiss** [comment]
File a separate issue, comments on closed issues are practically invisible except to people already subscribed, also this issue is unrelated - the battery issue has already been fixed and tools attached to workstations can suck in ~1MJ of energy for 1 craft, it's the recipes that might need tweaking

**Comment by RenechCDDA** [comment]
> An absurd example in the latest experimental build: -To boil 1L of water you need 17minutes and multi-coocker with 800 points of energy -To create pickaxe you need 7hours and electric forge with 40 points of energy So you need less energy to melt steel than to boil water.

Oh don't worry, the days of free energy for forging are dearly numbered.
